### DISCUSSION TITLE
arm64 detour throw 0xC000001D: Illegal Instruction. when call the real fn

### DISCUSSION THREAD

**Discussion by hereafter** [microsoft/Detours#276] [Q&A] [answered]
the detour works fine with x64 builds.
however when i switch it to arm64, i got 0xC000001D: Illegal Instruction.
when calling the RealFn

setup with
::DetourAttach(&(LPVOID&)RealFn, HookFn);

when calling the RealFn 
HookFn()
{
   RealFn();
}

unhandled exception   0xC000001D: Illegal Instruction.

Any idea what's up with that?

**Comment by cnbluefire** [comment]
I am hooking CreateWindowExW, and encounter the same problem

**Comment by LeoDavidson** [comment, marked as answer]
I don't know if this will be the issue, but make sure your replacement function uses the correct calling convention. ARM64 has two conventions, based on this OldNewThing post:

https://devblogs.microsoft.com/oldnewthing/20220823-00/?p=107041

_Quick! Click the link before Microsoft break every URL pointing to every developer blogs again, like they do every few months! :-)_

x64 only has one calling convention for everything, which I have seen mask mismatches until you try an x86 build (or an ARM64, I suspect).

**Reply by sylveon** [reply]
If you remove the cast to (LPVOID&) the template overload should catch these mistakes.

**Comment by kimjw0820** [comment]
I have the same problem.
Have you solved the x64 process issue on ARM64?
